### DISCUSSION TITLE
How do I make mockery return the same mock that has been called?

### DISCUSSION THREAD

**Discussion by floydjones1** [vektra/mockery#643] [Q&A]
I want my mock to return the same mock because the actual implementation returns a new pointer to the same struct with new internal variables. Any help would be appreciated!

![image](https://github.com/vektra/mockery/assets/43974214/b4608aeb-a425-4a76-9d00-63cb888fa7fa)

**Comment by floydjones1** [comment]
I get this error in my console:

![image](https://github.com/vektra/mockery/assets/43974214/f1021267-9608-4013-9d68-c212536fbd2b)

This is the code being tested which adds to help explain this scienerio better:

![image](https://github.com/vektra/mockery/assets/43974214/1082b8a4-dc17-42bd-b87d-defdb1f497c2)

**Comment by floydjones1** [comment]
This was an implementation error on my end closing 😅

**Reply by LandonTClipp** [reply]
Sounds good 👍🏻  This is a classic problem people have with chained method calls. Go does not support covariant return types. If your method returns `*models.IncidentHistoryDiffDB`, you _have_ to return that concrete type, not an interface or any other implementation of the interface. This is a limitation of the language unfortunately.

It kind of makes sense though... because downstream code might expect to have access to your type's attributes (if it's a struct). You can't access attributes of the underlying type in an interface without using reflection.
